internal/pkg/keeper/client.go

```diff
@@ -0,0 +1,353 @@
+//
+// Copyright (C) 2024 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package keeper
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v3/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v3/dtos"
+	dtoCommon "github.com/edgexfoundry/go-mod-core-contracts/v3/dtos/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v3/dtos/requests"
+	"github.com/edgexfoundry/go-mod-core-contracts/v3/dtos/responses"
+	"github.com/edgexfoundry/go-mod-core-contracts/v3/models"
+
+	"github.com/edgexfoundry/go-mod-registry/v3/pkg/types"
+)
+
+const (
+	defaultTimeout = 10 * time.Second
+)
+
+type keeperClient struct {
+	config              *types.Config
+	keeperUrl           string
+	serviceKey          string
+	serviceHost         string
+	servicePort         int
+	healthCheckRoute    string
+	healthCheckInterval string
+}
+
+// NewKeeperClient creates new Keeper Client. Service details are optional, not needed just for configuration, but required if registering
+func NewKeeperClient(registryConfig types.Config) (*keeperClient, error) {
+	client := keeperClient{
+		config:     &registryConfig,
+		serviceKey: registryConfig.ServiceKey,
+		keeperUrl:  registryConfig.GetRegistryUrl(),
+	}
+
+	// ServiceHost will be empty when client isn't registering the service
+	if registryConfig.ServiceHost != "" {
+		client.servicePort = registryConfig.ServicePort
+		client.serviceHost = registryConfig.ServiceHost
+		client.healthCheckRoute = registryConfig.CheckRoute
+		client.healthCheckInterval = registryConfig.CheckInterval
+	}
+
+	return &client, nil
+}
+
+// IsAlive simply checks if Keeper is up and running at the configured URL
+func (k *keeperClient) IsAlive() bool {
+	client := http.Client{Timeout: defaultTimeout}
+	resp, err := client.Get(k.keeperUrl + common.ApiPingRoute)
```

Comment:
use the contract client to Ping
https://github.com/edgexfoundry/go-mod-core-contracts/blob/b0286c308de37bd3d0f6157ee2560331f20d5af8/clients/interfaces/common.go#L19